pkg/networkservice/common/heal/gen.go

```diff
@@ -19,11 +19,30 @@
 package heal
 
 import (
+	"context"
 	"sync"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
 )
 
-//go:generate go-syncmap -output connection_map.gen.go -type connectionMap<string,connection>
-//go:generate go-syncmap -output client_connection_map.gen.go -type clientConnMap<string,clientConnInfo>
+type connectionInfo struct {
+	mut                   sync.Mutex
+	conn                  *networkservice.Connection
+	state                 connectionState
+	successVerificationCh chan struct{}
```

Comment:
Why we can't use `sync.Cond` as @Bolodya1997 suggested for simplifying the code?